runtime/compiler/z/codegen/S390PrivateLinkage.cpp

```diff
@@ -1742,10 +1742,11 @@ J9::Z::PrivateLinkage::buildNoPatchingIPIC(TR::Node *callNode, TR::RegisterDepen
 
    TR::RegisterDependencyConditions *preDeps = new (trHeapMemory()) TR::RegisterDependencyConditions(dependencies->getPreConditions(), NULL,
                                                                                                          dependencies->getAddCursorForPre(), 0, cg());
-   
-   TR::RegisterDependencyConditions *postDeps = new (trHeapMemory()) TR::RegisterDependencyConditions(NULL, dependencies->getPostConditions(),
-                                                                                                         0, dependencies->getAddCursorForPost() + numberOfAdditionalPostDeps, cg());
 
+	TR::RegisterDependencyConditions *postDeps = new (trHeapMemory()) TR::RegisterDependencyConditions(dependencies, 0, numberOfAdditionalPostDeps, cg());
```

Comment:
@fjeremic  Can I get a review please? I found out this when rebased my changes for Static and Special Direct calls for Read only cache and tested them out. 
This previous implementation of how we extract the Post dependencies from the dependencies and add room for new one was actually not working as it was just setting the cursor which is not what we want. To avoid the un-necessary object allocation for postDeps, so this way we just create the postDeps from the dependencies with room for new ones and set preDeps to 0.